Stellar-overlay.x

```diff
@@ -113,7 +113,11 @@ enum MessageType
 
     SEND_MORE = 16,
     FLOOD_ADVERT = 18,
-    FLOOD_DEMAND = 19
+    FLOOD_DEMAND = 19,
+    
+    // Configuration upgrades
+    GET_CONFIG_UPGRADE_SET = 20,
```

Comment:
I have a high level question about the [CAP](https://github.com/stellar/stellar-protocol/blob/master/core/cap-0046-09.md). Would it be possible to force validators to have the `ConfigUpgradeSets` locally, and avoid having to make these overlay changes? Validators would call the `upgrade` command with the full xdr, save it, but then vote on the hash. I'm still working out if this can be used maliciously.

Reply:
That's an interesting idea. My only concern is that if we ever allow fuzzy upgrade matching, then this won't work. E.g. if a validator wants to use its own calibration data and computes a coefficient a 1000, while another one computers a coefficient of 1001. I don't know if that's ever a good idea though.

Reply:
Yeah I don't see how adding this fuzzy matching complexity is a good idea. Is that flexibility helpful? Which setting would be applied to the network (1000 vs 1001 in your example)?. Upgrade values are meant to be determined offline. Once validator operators agree on what the value should be, they submit the settings to the network.

Reply:
The consensus could be done on e.g. the smallest hash. The reason would be so that validators could run the calibration themselves. Otherwise there needs to be some centralized distribution of the calibration data. That's probably fine in practice, but I'm not sure if we should eliminate this possibility. Maybe it's ok to leave it out at least for v1...

Reply:
Is there a CAP where I can read about this "calibration"? I don't think I have a good understanding of what you're referring to, and adding support for this in the overlay and consensus seems like overkill for what you get.